src/api/services/goals.service.js

```diff
@@ -30,3 +29,25 @@ exports.delete = async (goalId, userId) => {
     $pull: { createdGoals: goalId },
   }).exec();
 };
+
+exports.modifyMainGoal = async (mainGoalId, title) => {
+  return await MainGoal.findByIdAndUpdate(mainGoalId, {
+    $set: { title },
+  }).exec();
+};
+
+exports.modifySubGoal = async (subGoal, subGoalId, mainGoalId) => {
+  const { title, todos } = subGoal;
+  const mainGoal = await MainGoal.findById(mainGoalId).exec();
```

Comment:
위에서 코멘트한 것과 비슷한 맥락으로 아래처럼 쓸 수는 있을 것 같다고 생각했는데, 단순히 읽어오는 게 아니라 도큐먼트를 수정 후 저장해야 하니 현재 쓰신 형태가 맞을 것 같네요! 참고용으로만 달아두겠습니다.

```suggestion
  const subGoals = await MainGoal.findById(mainGoalId, "subGoals").exec();
```